File: src/module/buff/buff_protocol.erl
-module(buff_protocol).
-export([read/2, write/2]).
-include("buff.hrl").


read(11801, <<>>) ->
    {ok, []};

read(Code, Binary) ->
    {error, Code, Binary}.



write(11801, [List]) ->
    {ok, protocol:pack(11801, <<(length(List)):16, <<<<BuffId:32, EndTime:32, Overlap:16>> || #buff{buff_id = BuffId, end_time = EndTime, overlap = Overlap} <- List>>/binary>>)};

write(Code, Content) ->
    {error, Code, Content}.

File: script/make/protocol/json/BuffProtocol.js
let buffProtocol = {
    "read" : {
        "11801" : []
    },
    "write" : {
        "11801" : [
            {"name" : "list", "type" : "list", "comment" : "Buff列表", "explain" : [
                {"name" : "buffId", "type" : "u32", "comment" : "BuffID", "explain" : []},
                {"name" : "endTime", "type" : "u32", "comment" : "结束时间", "explain" : []},
                {"name" : "overlap", "type" : "u16", "comment" : "叠加数量", "explain" : []}
            ]}
        ]
    }
};